client/js/lounge.js

```diff
@@ -428,16 +490,20 @@ $(function() {
 		}
 
 		if (prevMsgTime.toDateString() !== msgTime.toDateString()) {
-			prevMsg.append(render("date-marker", {msgDate: msgTime}));
+			var parent = prevMsg.parent();
+			if (parent.hasClass("condensed")) {
+				prevMsg = parent;
+			}
+			prevMsg.after(render("date-marker", {msgDate: msgTime}));
```

Comment:
Nevermind, `after` is magic and works differently if you pass in a selector into it.